src/app/questionnaire-form-filler/questionnaire-form-filler.component.ts

```diff
@@ -43,28 +49,41 @@ export class QuestionnaireFormFillerComponent implements OnInit {
   }
 
   onChangeQuestionnaireResponse(response: fhir.r4.QuestionnaireResponse) {
+    this.questionnaireResponse = response;
+  }
+
+  onSubmit() {
+    this.log('submit questionnaire response', this.questionnaireResponse);
     if (this.questionnaire.extension) {
       for (const extension of this.questionnaire.extension) {
         if (
           'http://hl7.org/fhir/uv/sdc/StructureDefinition/sdc-questionnaire-targetStructureMap' ===
           extension.url
         ) {
           this.log('extraction');
-          const client = this.data.getFhirClient();
-          client
+          this.fhirKitClient
             .operation({
               name: 'extract',
               resourceType: 'QuestionnaireResponse',
-              input: response,
+              input: this.questionnaireResponse,
             })
             .then((data) => (this.extracted = data));
         }
       }
     }
-    this.questionnaireResponse = response;
+    return this.router.navigateByUrl('/');
```

Comment:
shouldn't we navigate back only when the submit action was finished?
And, for the other two functions as well, are the `return` statements necessary?

Reply:
This is the case in the newest version. The return is not really necessary but it removes the warnings about not doing anything with the promise.